src/ol/layer/Base.js

```diff
@@ -50,6 +51,10 @@ class BaseLayer extends BaseObject {
      * @type {Object<string, *>}
      */
     const properties = assign({}, options);
+    if (typeof options.properties === 'object') {
+      delete properties.properties;
+      assign(properties, options.properties);
+    }
```

Comment:
Would it make sense to reorder the assignments so that the values int the properties object will not overwrite the configured options? (Did not test this:)
```suggestion
    const properties = assign({}, options.properties, options);
    if (typeof options.properties === 'object') {
      if ('properties' in options.properties) {
        properties.properties = options.properties.properties;
      } else {
        delete properties.properties;
      }
    }
```

Reply:
The configured options are not affect by the original code, the `options` object is not affected. It actually is the other way around, a little further down the properties are overwritten with certain values.
Maybe we could add a note that it is not possible to set a property that also is an option.

Reply:
You are right, this is not needed.